src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -115,6 +119,58 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
                 // This method will return without triggering the build if the submission is not of type TEST.
                 triggerTemplateBuildIfTestCasesChanged(programmingExercise.getId(), result.getId());
             }
+
+            if (!isSolutionParticipation && !isTemplateParticipation) {
```

Comment:
We could factor this block out into an own method to keep the top level grading method short and simpler.

Reply:
Done